# Figures
Scripts used to generate all paper figures. 

## Dependencies
R packages ggplot2, gridExtra, data.table, scales

## Main Figures
Figure 1: Proportion of well-imputed (r2>0.8) biallelic SNVs by reference panel, study ancestry, and genotyping array

Figure 2: Heterozygous genotype concordance rates for rare variants by ancestry with TOPMed panel imputation

Figure 3: Regional variability in TOPMed reference panel imputation quality

Figure 4: Genomic features associated with TOPMed imputation quality of biallelic SNVs by ancestry

## Supplemental Figures
Figure S1: Effect of sample size on imputation quality metrics

Figure S2: Mean observed imputation r2 (aggregate r2) of biallelic SNVs by reference panel, study ancestry, and genotyping array

Figure S3: Proportion of well-imputed (r2>0.8) biallelic SNVs by reference panel using WGS-based and real Illumina OmniExpress arrays

Figure S4: Proportion of well-imputed (r2>0.8) biallelic SNVs by reference panel, genotyping array, and variant caller in Finnish study

Figure S5: Heterozygous genotype concordance rates for low-frequency variants by ancestry with TOPMed panel imputation

Figure S6: Heterozygous genotype concordance rates for common variants by ancestry with TOPMed panel imputation

Figure S7: Principal component analysis of WGS samples

Figure S8: Regional variability in imputation quality of common variants with the TOPMed reference panel by genotyping array and ancestry across all chromosomes

Figure S9: Repeat classes associated with TOPMed imputation quality of biallelic SNVs by ancestry

Figure S10: Genomic features associated with TOPMed imputation quality of biallelic SNVs by ancestry

Figure S11: Proportion of well-imputed (r2>0.8) biallelic SNVs by predicted functional impact and ancestry

Figure S12: Proportion of well-imputed (r2>0.8) variants by variant type, genotyping array, and ancestry with the TOPMed panel

Figure S13: Distribution of MAF for biallelic SNVs by ancestry

